oas3.yaml

```diff
@@ -0,0 +1,328 @@
+        - pets
+      summary: Mofify Pet (v1)
+      description: Modifies the information about a pet
+      operationId: pet_modify_v1
+      parameters:
+        - name: id
+          in: path
+          description: The unique ID of the idea you wish to retrieve.
+          required: true
+          schema:
+            type: string
+            pattern: ^[a-f0-9]{32}$
+      requestBody:
+        description: The changes you wish to make to the pet's information.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PetV1"
+            example:
+              status: adopted
+
+      responses:
+        200:
+          description: The newly modified pet's information.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PetV1"
+        404:
+          description: The pet could not be found.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        500:
+          description: An error occurred on the server.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+components:
+  schemas:
+    HealthV1:
+      required:
+        - status
+      type: object
+      properties:
+        status:
+          type: string
+          description: The current aggregate status of the service.
+          enum:
+            - Healthy
+            - Degraded
+            - Unhealthy
+        results:
+          type: object
+          additionalProperties:
+            type: object
+            required:
+              - status
+              - description
+            properties:
+              status:
+                type: string
+                description: The current status of the service's component.
+                enum:
+                  - Healthy
+                  - Degraded
+                  - Unhealthy
+              description:
+                type: string
+                description: A short description of this component's healthy.
+              data:
+                type: object
+                additionalProperties: true
+                properties: {}
+          readOnly: true
+          xml:
+            name: Status
+      xml:
+        name: Health
+      example:
+        ok: true
+
+    PetV1:
+      required:
+        - name
+        - kind
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: The unique ID of the pet in our database.
+          readOnly: true
+        name:
+          type: string
+          description: The pet's given name, used to call it for cuddles and petting.
+        status:
+          type: string
+          description: The current adoption status of the pet.
+          enum:
+            - available
+            - adopted
+            - unavailable
```

Comment:
What's the difference between adopted and unavailable?

Reply:
Well I figured that animals might not strictly fall into the "adopted" and "available for adoption" categories. For example, the shelter might house animals whose owners are travelling, or animals which are in the process of being adopted might be marked as "unavailable" while the paperwork is processed, even though they are still at the shelter and happy to accept visitors for walks, fetch etc. 🤔

Reply:
Ok, makes sense.  I guess some animals may also not be safe to adopt, but they don't want to put them down, so they may be indefinite tenants.  Would it make sense to use statuses such as available-for-adoption, adoptable, not-available-for-adoption, not-adoptable, etc. to clarify it further?

Reply:
Yeah, we could definitely expand the taxonomy of terms here - I wanted to be a bit obtuse on purpose to avoid the "put down" aspect of things since our animal shelter is meant to be nothing but rainbows and butterflies. Another idea might be the inclusion of a `statusReason` field which allows you to add a short description of why the animal is in a particular state (and allows for more comprehensive, user defined, reasons for an animal to be in a particular state - such as "too young to adopt", "in quarantine", "my favourite lapdog" etc).